hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -1092,6 +1092,10 @@ class HoodieSparkSqlWriterInternal {
       && mergedParams.getOrElse(DataSourceWriteOptions.TABLE_TYPE.key, COPY_ON_WRITE.name) == MERGE_ON_READ.name) {
       mergedParams.put(HoodieTableConfig.DROP_PARTITION_COLUMNS.key, "false")
     }
+    // use meta sync database to fill hoodie.table.name if it not sets
+    if (!mergedParams.contains(HoodieTableConfig.DATABASE_NAME.key()) && mergedParams.contains(HoodieSyncConfig.META_SYNC_DATABASE_NAME.key())) {
```

Comment:
Are you saying `hoodie.database.name` ?

Reply:
@danny0405 Yes, I updated the comments just now.

Reply:
It looks like you are fixing the meta sync config, maybe just a UT for the spark write config.